locust/webui/README.md

```diff
@@ -0,0 +1,92 @@
+# Locust UI
+
+The Locust UI is used for viewing stats, reports, and information on your current Locust test from the browser.
+
+## Locust UI as a Library
+
+The Locust UI may be extended to fit your needs. If you only need limited extensibility, you may do so in your Locustfile, see the [extend_web_ui example](https://github.com/locustio/locust/blob/master/examples/extend_web_ui.py). 
+
+However, you may want to further extend certain functionalities. To do so, you may replace the default Locust UI with your own React application. Start by installing the locust-ui:
+```sh
+npm install locust-ui
+```
+or
+```sh
+yarn add locust-ui
+```
+
+## Usage
+
+```js
+import LocustUi from "locust-ui";
+
+<LocustUi<"content-length", "content_length">
+    extendedTabs={[
+        {
+            title: "Content Length",
+            key: "content-length",
+        },
+    ]}
+    extendedTables={[
+        {
+        key: "content-length",
+        structure: [
+            { key: "name", title: "Name" },
+            { key: "content_length", title: "Total content length" },
+        ],
+        },
+    ]}
+    extendedReports={[
+        {
+            href: "/content-length/csv",
+            title: "Download content length statistics CSV",
+        },
+    ]}
+    extendedStats={[
+        {
+            key: "content-length",
+            data: [{ name: "/", safeName: "/", content_length: "123" }],
+        },
+    ]}
+/>
+```
+The Locust UI package exports a Redux store. By wrapping your App in a Redux provider, you may be able to read Locust data from state:
+```js
+import { locustStore } from "locust-webui";
+import { Provider } from "react-redux";
+
+ReactDOM.createRoot(document.getElementById("root")!).render(
+    <React.StrictMode>
+        <Provider store={locustStore}>
+            <App />
+        </Provider>
+    </React.StrictMode>
+);
+```
+
+By default, the extended tabs will display the provided data in a table. However you may render any React component in the tab:
+```js
+import { IRootState } from "locust-webui";
+import { useSelector } from "react-redux";
+
+function MyCustomTab() {
+    const extendedStats = useSelector(
+        ({ ui: { extendedStats } }: IRootState) => extendedStats
+    );
+
+    return <div>{JSON.stringify(extendedStats)}</div>;
+}
+
+const extendedTabs = {[
+    {
+        title: "Content Length",
+        key: "content-length",
+        component: MyCustomTab
+    },
+]};
+```
+
+## Locustfile
+
+In order for Locust to use your frontend as the UI, you will have to configure your locustfile to point to the build output folder. See the [locust_ui_as_a_module example](https://github.com/locustio/locust/blob/master/examples/locust_ui_as_a_module.py).
```

Comment:
This sentence doesnt make a ton of sense IMO. Maybe just say something about "configure Locust to point to your own React build output"?